tests/httpLink.ts

```diff
@@ -0,0 +1,215 @@
+import { assert, expect } from 'chai';
+import * as sinon from 'sinon';
+import HttpLink from '../src/httpLink';
+
+import * as Links from '../src/link';
+import { ApolloLink } from '../src/link';
+
+import { createApolloFetch } from 'apollo-fetch';
+
+import { print } from 'graphql';
+import gql from 'graphql-tag';
+import * as fetchMock from 'fetch-mock';
+
+const sampleQuery = gql`
+query SampleQuery {
+  stub{
+    id
+  }
+}
+`;
+
+const sampleMutation = gql`
+mutation SampleMutation {
+  stub(param: "value"){
+    id
+  }
+}
+`;
+
+describe('HttpLink', () => {
+  const data = {hello: 'world', method: 'POST'};
+  const mockError = { throws: new TypeError('mock me') };
+
+  let subscriber;
+
+  beforeEach(() => {
+    fetchMock.post('begin:data', data);
+    fetchMock.post('begin:error', mockError);
+
+    const next = sinon.spy();
+    const error = sinon.spy();
+    const complete = sinon.spy();
+
+    subscriber = {
+      next,
+      error,
+      complete,
+    };
+  });
+
+  afterEach(() => {
+    fetchMock.restore();
+  });
+
+  it('raises warning if called with concat', () => {
+    const link = ApolloLink.from([new HttpLink()]);
+    const _warn = console.warn;
+    console.warn = warning => assert.property(warning, 'message');
+    assert.deepEqual(link.concat((operation, forward) => forward(operation)), link);
+    console.warn = _warn;
+  });
+
+  it('does not need any constructor arguments', () => {
+    assert.doesNotThrow(() => new HttpLink());
+  });
+
+  it('calls next and then complete', (done) => {
+    const next = sinon.spy();
+    const link = new HttpLink({uri: 'data'});
+    const observable = link.request({
```

Comment:
For all of these tests, I think we should be using `execute(link, operation)` instead of the `.request`. As link grows, I can see things being done during execution that each individual ink wouldn't need to ensure in its request (for instance mutation of the operation). 

I don't think `request` is really meant to be used publically on the link, instead it is used by the library itself to build the execution chain